Classes/common/HTTP/CDTURLSessionTask.m

```diff
@@ -84,12 +92,15 @@ - (void)cancel
     if (t) {
         [t cancel];
     }
+    _finished = YES;
```

Comment:
I don't think this method does what it is supposed to anymore, I can't see somewhere where the `NSURLSessionTask` get set, so calling `-cancel` results in a no op. The test cases don't take account for this because they assume the `NSURLSessionTask` is created.